detector/filecontent_detector.go

```diff
@@ -31,66 +31,126 @@ func (fc *FileContentDetector) AggressiveMode() *FileContentDetector {
 	return fc
 }
 
-func (fc *FileContentDetector) Test(additions []gitrepo.Addition, ignoreConfig TalismanRCIgnore, result *DetectionResults) {
-	cc := NewChecksumCompare(additions, ignoreConfig)
-	for _, addition := range additions {
-		if ignoreConfig.Deny(addition, "filecontent") || cc.IsScanNotRequired(addition) {
-			log.WithFields(log.Fields{
-				"filePath": addition.Path,
-			}).Info("Ignoring addition as it was specified to be ignored.")
-			result.Ignore(addition.Path, "filecontent")
-			continue
-		}
-
-		if string(addition.Name) == DefaultRCFileName {
-			re := regexp.MustCompile(`(?i)checksum[ \t]*:[ \t]*[0-9a-fA-F]+`)
-			content := re.ReplaceAllString(string(addition.Data), "")
-			data := []byte(content)
-			addition.Data = data
-		}
+type contentType int
 
-		base64Results := fc.detectFile(addition.Data, checkBase64)
-		fillBase46DetectionResults(base64Results, addition, result)
+const (
+	base64Content contentType = iota
+	hexContent
+	creditCardContent
+)
 
-		hexResults := fc.detectFile(addition.Data, checkHex)
-		fillHexDetectionResults(hexResults, addition, result)
+func (ct contentType) getInfo() string {
+	switch ct {
+	case base64Content:
+		return "Failing file as it contains a base64 encoded text."
+	case hexContent:
+		return "Failing file as it contains a hex encoded text."
+	case creditCardContent:
+		return "Failing file as it contains a potential credit card number."
+	}
+	return ""
+}
 
-		creditCardResults := fc.detectFile(addition.Data, checkCreditCardNumber)
-		fillCreditCardDetectionResults(creditCardResults, addition, result)
+func (ct contentType) getOutput() string {
```

Comment:
Can this method be called getMessageFormat ?

Reply:
Done. Thanks